### PR TITLE
fix #19 target modern JS modules

### LINKED ISSUE #19: Target Modern Runtimes for JS SDK
The current JS SDK is distributed to NPM with ES5 target. This is very antiquated method as argued [here](https://web.dev/articles/publish-modern-javascript) and [here](https://jasonformat.com/enabling-modern-js-on-npm).

Almost all the applications which use TastyTrade SDK will likely also use [DXLink Websocket Client](https://github.com/dxFeed/dxLink/tree/main/dxlink-javascript/dxlink-websocket-client) which is also authored as modern javascript module on NPM - See [reference tsconfig.json](https://github.com/dxFeed/dxLink/blob/main/dxlink-javascript/tsconfig.json)

Can we target modern runtimes for JS SDK for Tasty Trade APIs?

If the is willing to support this feature, I can even attempt to submit a PR to that effect.

### PR DESCRIPTION
**Pull request by osdevisnot**
In addition to targeting modern js syntax code we should target modern JS modules too.

### CODE REVIEW

**Comment by osdevisnot** [comment]
@dmoss18 after this MR is merged, we should publish a new major version to NPM.

File: lib/account-streamer.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import WebSocket from 'isomorphic-ws'
 import _ from 'lodash'
-import type { JsonMap, JsonValue } from './utils/json-util'
-import { JsonBuilder } from './utils/json-util'
-import TastytradeSession from './models/tastytrade-session'
-import { MinTlsVersion } from './utils/constants'
+import type { JsonMap, JsonValue } from './utils/json-util.js'
```
Comment (dmoss18) [inline lib/account-streamer.ts:3]: Can you revert all the `.js` suffixes? This is a typescript project. The javascript files should all get compiled into the `dist` folder.

**Review by dmoss18 — CHANGES_REQUESTED** [review]

**Comment by dmoss18** [comment]
> @dmoss18 after this MR is merged, we should publish a new major version to NPM.

A major version means there would be breaking changes. Have we had any of those? I'd expect this to be a minor release.

File: lib/account-streamer.ts (hunk 1 above)
Comment (osdevisnot) [reply lib/account-streamer.ts:3]: TS compiler does not rewrite import paths. This is something the TS team has been adamant about. Omitting `.js` extension here will result in an error for all node.js users with ERR_MODULE_NOT_FOUND for the compiled files in dist. See this stack overflow post for more details on this topic - https://stackoverflow.com/a/62626938

Ideally, we would want to adopt a bundler to bundle all the files in dist if we wish to keep extension less imports in src. This however is much larger change!

**Comment by osdevisnot** [comment]
> > @dmoss18 after this MR is merged, we should publish a new major version to NPM.
> 
> A major version means there would be breaking changes. Have we had any of those? I'd expect this to be a minor release.

I would assume a change in the distribution format `cjs -> esm` as a major change.

File: lib/account-streamer.ts (hunk 1 above)
Comment (dmoss18) [reply lib/account-streamer.ts:3]: Can you add some steps to reproduce the issue you are describing?

Comment (dmoss18) [reply lib/account-streamer.ts:3]: @osdevisnot I see the issue. I'll add a couple things to the tsconfig and we can merge this.

**Review by dmoss18 — APPROVED** [review]

**Pull request merged**